internal/command/machine/run.go

```diff
@@ -514,24 +517,68 @@ func determineMachineConfig(ctx context.Context, initialMachineConf api.MachineC
 			return
 		}
 		machineConf.Guest = guest
+
```

Comment:
This seems like a lot to keep track of and potentially easy to miss when adding new fields. I feel like I’ve seen some library that can show a JSON diff in a meaningful way and wonder if that could be easier to maintain while still keeping the valuable diff.

Reply:
Perhaps https://github.com/google/go-cmp?

Reply:
Oh! we were looking into https://github.com/nsf/jsondiff but the library you mentioned looks cool. I like that the diff shows (+) and (-) so we don't have to format the diff ourselves 

https://play.golang.com/p/SQfAG9nPsoN